### PR TITLE
[fix](nereids) fix bug when sum0 distinct appear in cte

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #45209

Problem Summary:
```code
with tmp as (select * from t1) select * from (select sum0(distinct b) from tmp) t cross join (select sum0(distinct a) from tmp) tt
```
before this pr, this sql report error:
VSlotRef xxx have invalid slot id.
This pr fix this problem.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]
add related pr

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java [hunk 1]
```diff
@@ -2076,7 +2079,8 @@ private boolean couldConvertToMulti(LogicalAggregate<? extends Plan> aggregate)
             if (!func.isDistinct()) {
                 continue;
             }
-            if (!(func instanceof Count || func instanceof Sum || func instanceof GroupConcat)) {
+            if (!(func instanceof Count || func instanceof Sum || func instanceof GroupConcat
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:2082]: how about add a new Interface "MultiDistinct", and let these Agg functions implement it?

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java (hunk 1 above)
Comment (feiniaofeiafei) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/AggregateStrategies.java:2082]: Yes, an interface is added in #45209, I will use that interface  in that pr.

**Pull request merged**